projects/storefrontlib/src/events/page/page-event.builder.ts

```diff
@@ -28,13 +30,19 @@ export class PageEventBuilder {
   }
 
   protected buildPageEvent(): Observable<PageEvent> {
-    return this.getNavigatedEvent().pipe(
-      map((state) =>
+    const navigation$ = this.getNavigatedEvent();
+    const pageTitle$ = this.contentPageMetaResolver.resolveTitle();
+
+    return zip(navigation$, pageTitle$).pipe(
```

Comment:
I'm not a king here, but every time i tried to use zip, there was a better approach.  
What about a `switchMap` from the navigation$ stream to the `PageMeta` stream? with mapping the data.